dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v36/V2_36_4__Remove_duplicate_mappings_from_categoryoption_to_usergroups.java

```diff
@@ -0,0 +1,176 @@
+                do
+                {       
+                    duplicateCatOptUsrgrp = new DuplicateCategoryOptionUserGroupWithCount();
+                    duplicateCatOptUsrgrp.setName( rs.getString( "name" ) );
+                    duplicateCatOptUsrgrp.setCategoryOptionId( rs.getLong( CATEGORYOPTIONID ) );
+                    catOptIds.add( rs.getString( CATEGORYOPTIONID )  );
+                    duplicateCatOptUsrgrp.setUserGroupId( rs.getLong( USERGROUPID ) );
+                    userGroupIds.add( rs.getString( USERGROUPID ) );
+                    duplicateCatOptUsrgrp.setCount( rs.getInt( "count" ) );
+                    totalCount = totalCount + rs.getInt( "count" );
+                    duplicateCatOptUsrgrps.add( duplicateCatOptUsrgrp );
+                }
+                while ( rs.next() );
+            }
+        }
+        
+        Set<Pair<String,String>> catOptUsrGroupPairs = new HashSet<>();
+        Set<String> deleteUserGroupAccessIds = new HashSet<>();
+        
+        String catOptIdsCommaSeparated = StringUtils.join( catOptIds, "," );
+        String userGroupIdsCommaSeparated = StringUtils.join( userGroupIds, "," );
+      
+        // 2. Get ordered access strings for each categoryoptionid and  usergroupid combination 
+        try ( Statement stmt = context.getConnection().createStatement();
+            ResultSet rs = stmt.executeQuery( String.format( GET_ACCESS_STRING_FOR_CO_UG_COMBO, catOptIdsCommaSeparated, userGroupIdsCommaSeparated ) ); )
+        {
+            Pair<String,String> currentPair = null;
+            while ( rs.next() )
+            {
+                currentPair = Pair.of( rs.getString( CATEGORYOPTIONID ), rs.getString( USERGROUPID ) );
+                if ( catOptUsrGroupPairs.contains( currentPair ) )
+                {
+                    deleteUserGroupAccessIds.add( rs.getString( "usergroupaccessid" ) );
+                }
+                else
+                {
+                    catOptUsrGroupPairs.add( currentPair );
+                }
+            }
+
+        }
+        log.info( "Total identified duplicate userGroupAccessIds to delete : " + deleteUserGroupAccessIds.size() );
+        log.info( "Total catOpts duplicated in mappings: " + duplicateCatOptUsrgrps.size() );
+        log.info( "Total impacted rows in usergroupaccess mapping tables: " + totalCount );
+      
+        String usrGrpAccessIdsCommaSeparated = StringUtils.join( deleteUserGroupAccessIds, "," );
+        
+        try ( Statement stmt = context.getConnection().createStatement(); )
+        {
+            //Delete redundant usergroupaccessid from dataelementcategoryoptionusergroupaccesses table
+            int deletedCougaCount = stmt.executeUpdate( String.format( DELETE_CATOPT_USRGRP_ACCESS, usrGrpAccessIdsCommaSeparated ) );
+            log.info( "Deleted userGroupAccessIds from dataelementcategoryoptionusergroupaccesses table : " + deletedCougaCount );
```

Comment:
looks fine to me.
this message is displaying count so it can be changed to for example
`Number of userGroupAccessIds deleted from dataelementcategoryoptionusergroupaccesses table :`